haskell_scotty/ClientRepo.hs

```diff
@@ -0,0 +1,36 @@
+{-# LANGUAGE OverloadedStrings #-}
+module ClientRepo where
+
+import Client
+import DBConfig
+import Database.MySQL.Simple
+import Database.MySQL.Simple.Result
+import Database.MySQL.Simple.QueryResults
+
+all :: IO [Client]
+all = do
+  DBConfig user pass database <- readDBConfig
+  conn <- connect defaultConnectInfo {
+            connectUser = user,
+            connectPassword = pass,
+            connectDatabase = database }
+  clients <- query_ conn "SELECT * FROM clients"
+  return clients
+
+instance QueryResults Client where
+  convertResults [fa,fb,fc,fd,fe,ff,fg,fh,fi,fj,fk,fl] [va,vb,vc,vd,ve,vf,vg,vh,vi,vj,vk,vl] =
+    Client a b c d e f g h i j k l 
+      where a = convert fa va
+            b = convert fb vb
+            c = convert fc vc
+            d = convert fd vd
+            e = convert fe ve
+            f = convert ff vf
+            g = convert fg vg
+            h = convert fh vh
+            i = convert fi vi
+            j = convert fj vj
+            k = convert fk vk
+            l = convert fl vl
```

Comment:
```haskell
convertFunction pair =
  convert (fst pair) (snd pair)
 
instance QueryResults Client where
  convertResults fields values =
    let v = map convertFunction (Data.List.zip fields values) in
      Client (!! v 0) (!! v 1) (!! v 2) (!! v 3) (!! v 4) (!! v 5) (!! v 6) (!! v 7) (!! v 8) (!! v 9) (!! v 10) (!! v 11)
```

This was as far as I got before I remembered that lists can't contain heterogeneous types, found out you cant map a list to a tuple, then subsequently bashed my face into the keyboard for ten minutes.